internal/resolver/dns/dns_resolver_test.go

```diff
@@ -126,8 +127,11 @@ func (*testResolver) LookupTXT(ctx context.Context, host string) ([]string, erro
 
 func replaceNetFunc(ch chan struct{}) func() {
```

Comment:
I hate to ask you to change stuff unrelated to what you are trying to fix here. But `replaceNetFunc` is a super unintuitive name for a function which replaces the `defaultResolver` with a `testResolver`.

Something like `overrideDefaultResolver()` or `overrideNetResolver()` with a comment on the function describing what it is doing.

Reply:
Also, while you are here and since the channel in the `testResolver` has been changed to a `testutils.Channel`, the `ch` argument to this function no longer makes sense.

You could write this func as follows:
```golang
// overrideDefaultResolver overrides the defaultResolver used by the code with an instance of
// the testResolver. pushOnLookup controls whether the testResolver created here pushes
// lookupHost events on its channel.
func overrideDefaultResolver(pushOnLookup bool) func() {
	oldResolver := defaultResolver

	var ch *testutils.Channel
	if pushOnLookup {
		ch = testutils.NewChannel()
	}
	defaultResolver = &testResolver{ch: ch}

	return func() {
		defaultResolver = oldResolver
	}
}
```

Reply:
I agree. ReplaceNetFunc() is a weird name. Also yeah makes sense thanks for the idea about switching the argument. I didn't like passing a channel either. Switching now.

Reply:
Switched. Thanks :).